includes/class-amp-story-post-type.php

```diff
@@ -174,2197 +102,179 @@ public static function register() {
-					}
-				}
-			}
-
-			if ( $found ) {
-				if ( '' !== $css ) {
-					$css .= ';';
-				}
-
-				$css .= $css_item;
+	public static function replace_editor( $replace, $post ) {
+		if ( self::POST_TYPE_SLUG === get_post_type( $post ) ) {
+			$replace = true;
+			// In lieu of an action being available to actually load the replacement editor, include it here
+			// after the current_screen action has occurred because the replace_editor filter fires twice.
+			if ( did_action( 'current_screen' ) ) {
+				require_once AMP__DIR__ . '/includes/edit-story.php';
 			}
 		}
-
-		return $css;
+		return $replace;
 	}
 
 	/**
-	 * Filters the response before executing any REST API callbacks.
-	 *
-	 * Temporarily modifies post content during saving in a way that KSES
-	 * does not strip actually valid CSS from post content, making block content invalid.
-	 *
-	 * @todo Remove once core has better CSS parsing.
-	 *
-	 * @link https://core.trac.wordpress.org/ticket/37134
-	 *
-	 * @param WP_HTTP_Response|WP_Error $response Result to send to the client. Usually a WP_REST_Response or WP_Error.
-	 * @param array                     $handler  Route handler used for the request.
-	 * @param WP_REST_Request           $request  Request used to generate the response.
-	 *
-	 * @return WP_HTTP_Response|WP_Error The filtered response.
+	 * Enqueue scripts for the block editor.
 	 */
-	public static function filter_rest_request_for_kses( $response, $handler, $request ) {
+	public static function admin_enqueue_scripts() {
+		$screen = get_current_screen();
 
-		// Short-circuit since this is relevant only for users without unfiltered_html capability.
-		if ( current_user_can( 'unfiltered_html' ) ) {
-			return $response;
+		if ( ! $screen instanceof \WP_Screen ) {
+			return;
 		}
 
-		if ( is_wp_error( $response ) ) {
-			return $response;
+		if ( self::POST_TYPE_SLUG !== $screen->post_type ) {
+			return;
 		}
 
-		$obj  = get_post_type_object( self::POST_TYPE_SLUG );
-		$slug = ! empty( $obj->rest_base ) ? $obj->rest_base : $obj->name;
+		$asset_file   = AMP__DIR__ . '/assets/js/' . self::AMP_STORIES_SCRIPT_HANDLE . '.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
 
-		$editable_request_methods = array_map( 'trim', explode( ',', WP_REST_Server::EDITABLE ) );
+		wp_enqueue_script(
+			self::AMP_STORIES_SCRIPT_HANDLE,
+			amp_get_asset_url( 'js/' . self::AMP_STORIES_SCRIPT_HANDLE . '.js' ),
+			$dependencies,
+			$version,
+			false
+		);
 
-		if ( ! in_array( $request->get_method(), $editable_request_methods, true ) || ! preg_match( "#^/wp/v2/{$slug}/#s", $request->get_route() ) ) {
-			return $response;
-		}
+		if ( function_exists( 'wp_set_script_translations' ) ) {
+			wp_set_script_translations( 'amp-editor-story-blocks-build', 'amp' );
+		} elseif ( function_exists( 'wp_get_jed_locale_data' ) || function_exists( 'gutenberg_get_jed_locale_data' ) ) {
```

Comment:
Do we still need all this? Since we're not using the blocks build file anymore.

(FI: `gutenberg_get_jed_locale_data` was to support times when Gutenberg wasn't included in WP yet, I guess that's irrelevant for the new editor, anyway)

Reply:
Should we remove `wp_get_jed_locale_data`?

Reply:
It looks like we can generally remove the whole `if` condition since it doesn't seem like we're using the `amp-editor-story-blocks-build` file at all.

Reply:
This function only exists in old Gutenberg running on WP 4.9. So this can definitely be removed.

Reply:
Removed here - https://github.com/ampproject/amp-wp/pull/3839/commits/fcffc0828d2eeaec6bf22ec041e16649f5895cb1